Added workflows

SuperNinja-4965 opened the pull request:
Added build and test workflow for building and testing the code. (useful to see if code will build and a pr hasnt broken the code) 
Added a codeql workflow for code analysis (might be useful).

benjaminkay93 commented:
![current](https://user-images.githubusercontent.com/10714808/61890707-ebf36080-aeff-11e9-9ae8-d7c274c34430.png)
![step_1](https://user-images.githubusercontent.com/10714808/61890714-ef86e780-aeff-11e9-9049-850b67f3a143.png)
![step_2](https://user-images.githubusercontent.com/10714808/61890716-f0b81480-aeff-11e9-8d9d-8ee9a0d57994.png)
![step_3](https://user-images.githubusercontent.com/10714808/61890717-f150ab00-aeff-11e9-91ea-6d1799b1e49a.png)
![step_4](https://user-images.githubusercontent.com/10714808/61890723-f57cc880-aeff-11e9-8d52-4867286b5678.png)

tjenkinson commented:
Nice. We need to make sure we handle the old URL structure too so that old links don't break

benjaminkay93 commented:
@tjenkinson those images were just commented so I could just swap to my mac 🤣. I've written the whole document upon the strategy to migrate that was discussed last night, all be it with different numbers to how @m3lnic wrote it. 

This is purely to decouple the front end, the links stuffwe will have to handle, not sure how though. It would be really handy to have a complete archetecture diagram of what exists at the moment, and how things talk to one another, as there will be a bit were we have to redirect wowza and other vms to point to the api backend instead of www backend

m3lnic commented:
I mean, I would be happy putting the new front end on the new API VM that we are requesting. But it would mean putting in place infrastructure that would be removed later on. It would be nice to have the front and back ends on different servers too. Make for easier management of things down the road imo.

moschopsuk commented on docs/migration.md:
```diff
@@ -0,0 +1,39 @@
+# Migration Plan
+
+The current LA1TV website offering is a monolith application, there is no archectural seperation of concerns between the components that form the website. This has lead to a situation where the website has been poorly maintained by the membership over the last 3 years and it has slowly become more and more of a risk to the continued operation of LA1TV.
```
architectural separation

benjaminkay93 replied:
```suggestion
The current LA1TV website offering is a monolith application, there is no architectural separation of concerns between the components that form the website. This has lead to a situation where the website has been poorly maintained by the membership over the last 3 years and it has slowly become more and more of a risk to the continued operation of LA1TV.
```

moschopsuk commented on docs/migration.md:
```diff
@@ -0,0 +1,39 @@
+
+The current LA1TV website offering is a monolith application, there is no archectural seperation of concerns between the components that form the website. This has lead to a situation where the website has been poorly maintained by the membership over the last 3 years and it has slowly become more and more of a risk to the continued operation of LA1TV.
+
+The current site however, is the most featurerful out of any other student television station's and in it's abilities rivals any broadcasting companys technological abilities except that of major broadcasters such as the BBC. This functionality is wanted to be maintained and improved upon with other features such as: live blogging; reactions; more share features and improved onward journies.
```
featureful, journeys

benjaminkay93 replied:
```suggestion
The current site, however, is the most featureful out of any other student television station's and in its abilities rivals any broadcasting company's technological abilities except that of major broadcasters such as the BBC. This functionality is wanted to be maintained and improved upon with other features such as live blogging; reactions; more share features and improved onward journeys.
```

moschopsuk commented on docs/migration.md:
```diff
@@ -0,0 +1,39 @@
+
+The current site however, is the most featurerful out of any other student television station's and in it's abilities rivals any broadcasting companys technological abilities except that of major broadcasters such as the BBC. This functionality is wanted to be maintained and improved upon with other features such as: live blogging; reactions; more share features and improved onward journies.
+
+To achieve this without any major regression in capabilities a soloution must be found to breaking down the monolith, something that has proved itself to be hard to achieve. Outlined below is a series of steps that could effectively decouple the frontend of the LA1TV website safely with no regression in capabilities on the backend, and only minor front end loses of functionality.
```
solution

benjaminkay93 replied:
```suggestion
To achieve this without any major regression in capabilities a solution must be found to breaking down the monolith, something that has proved itself to be hard to achieve. Outlined below is a series of steps that could effectively decouple the frontend of the LA1TV website safely with no regression in capabilities on the backend, and only minor front end lose of functionality.
```

moschopsuk commented on docs/migration.md:
```diff
@@ -0,0 +1,39 @@
+
+## Step 3
+
+Once the new www frontend has been developed to a point where LA1TV are confident in its abilities, we can then deploy it over the top of the current www frontent, removing the old service from the view of the audience.
```
frontend

moschopsuk commented on docs/migration.md:
```diff
@@ -0,0 +1,39 @@
+
+## Step 4
+
+Once we have detached the audience from the old frontend, we can then remove that code from the backend service. removing a large part of the comlpexity that currently exsists on the current website.
```
complexity  exists

benjaminkay93 replied:
```suggestion
Once we have detached the audience from the old frontend, we can then remove that code from the backend service. removing a large part of the complexity exists on the current website.
```

moschopsuk commented on docs/migration.md:
```diff
@@ -0,0 +1,39 @@
+
+## Post frontend migration
+
+The frontend will now be isolated from the rest of the codebase, allowing LA1TV to actively start developing the frontend to suit their wants an needs rather than face the monolith that was the old website. The frontend can be built independantly, using the latest in web design practises giving skills to members that they would not of been able to gain on the old stack. It can also be built simpl ...
```
independently

benjaminkay93 replied:
```suggestion
The frontend will now be isolated from the rest of the codebase, allowing LA1TV to actively start developing the frontend to suit their wants an needs rather than face the monolith that was the old website. The frontend can be built independently, using the latest in web design practises giving skills to members that they would not of been able to gain on the old stack. It can also be built simpler, having a much lower barrier to entry than the old php site.
```

benjaminkay93 commented:
Thanks all, raised #8 to further discuss deployment, be good to brain dump on there :)

SuperNinja-4965 committed:
Added workflows
Added build and test workflow for building and testing the code. (useful to see if code will build and a pr hasnt broken the code)
Added a codeql workflow for code analysis (might be useful).

SuperNinja-4965 committed:
Rename workflow, Use example config instead of echo -e

SuperNinja-4965 committed:
Add reportCoverage command to build

SuperNinja-4965 committed:
Added build commands to codeql
